src/TrackingProviders/ImpactTrackingProvider.php

```diff
@@ -0,0 +1,220 @@
+            "
+            . self::$bodyTop;
+    }
+
+    /**
+     * @param array $products
+     * @param $transactionId
+     * @param $revenue
+     * @param $tax
+     * @param $shipping
+     * @param string $currency
+     */
+    public static function trackTransaction(
+        array $products,
+        $transactionId,
+        $revenue,
+        $tax,
+        $shipping,
+        $paymentType = '',
+        $promoCode = '',
+        $currency = 'USD'
+    )
+    {
+        $tagActionTrackerId = config('railanalytics.' . env('APP_ENV') . '.providers.impact.tag-action-tracker-id');
+        $signUpActionTrackerId = config('railanalytics.' . env('APP_ENV') . '.providers.impact.sign-up-action-tracker-id');
+
+        $status = "";
+        if ($paymentType == "initial_order") {
+            $status = "New";
+        } elseif ($paymentType == "subscription_renewal") {
+            $status = "Returning";
+        }
+
+        $totalDiscount = 0;
+        $jsonProductsArray = [];
+        foreach ($products as $product) {
+            $totalDiscount =+ $product['discount'];
+            $jsonProductsArray[] = "
+                            {
+                                subTotal: " . $product['quantity'] * $product['value']. ",
+                                category: \"" . $product['category'] . "\",
+                                sku: \"" . $product['sku'] . "\",
+                                quantity: " . $product['quantity'] . ",
+                                name: \"" . $product['name'] . "\",
+                            },";
+
+        }
+        $output =
+            "
+                <script type='text/javascript'>
+                    ire('trackConversion', $signUpActionTrackerId, {
+                        orderId: '" . $transactionId . "',
+                        customerId: '" . self::$customerId . "',
+                        customerEmail: '" . self::$customerEmail . "'
+                    },
+                    {
+                        verifySiteDefinitionMatch:true
+                    }
+                    );
+
+                    ire('trackConversion', $tagActionTrackerId, {
+                        orderId: '" . $transactionId . "',
+                        customerId: '" . self::$customerId . "',
+                        customerEmail: '" . self::$customerEmail . "',
+                        customerStatus: '" . $status . "',
+                        currencyCode: '" . $currency . "',
+                        orderPromoCode: '" . $promoCode . "',
+                        orderDiscount: " . $totalDiscount . ",
+                        items: [";
+        $output .= implode(" ", $jsonProductsArray);
+        $output .=
+                    "
+                        ],
+                    });
+            " .
+            "
+                </script>
+            ";
+
+        self::$headBottom .= $output;
+
+    }
+
+    public static function trackTransactionAPI( array $products,
```

Comment:
@mirceamartsoft - Is this function still nessecary? I could not find any usages.